src/routers/products.router.ts

```diff
@@ -66,25 +72,53 @@ export class ProductRouter extends BaseModelRouter<IProduct> {
         );
 
         // ----------------------------------- POST -----------------------------------\\
+/*
+        this.router.post('/', async (req: Request, res: Response) => {
+            try {
+                const newProducts = req.body as IProduct[];
+                if (!Array.isArray(newProducts)) {
+                    return res.status(400).json({ error: 'Request body must be an array' });
+                }
+
+                const result = await this.model.insertMany(newProducts);
+
+                if (result) {
+                    res.status(201).json(result); // Ensure response is JSON formatted
+                } else {
+                    res.status(500).json({ error: 'Failed to create new products' });
+                }
+            } catch (error) {
+                console.error('Error creating products:', error);
+                res.status(400).json({ error: error.message });
+            }
+        });
+
+*/
 
         this.router.post('/', async (req: Request, res: Response) => {
             try {
-                const newProduct = req.body as IProduct;
-                const result = await this.model.insertMany(newProduct);
+                const newProducts = req.body as IProduct[];
+                if (!Array.isArray(newProducts)) {
+                    return res.status(400).json({ error: 'Request body must be an array of products' });
+                }
 
+                const result = await this.model.insertMany(newProducts);
+                console.log(result)
```

Comment:
what is the difference between this and the commented out code, remove commented code

Reply:
remove console log

Reply:
using .send, you should be able to assert a return value of string.

Reply:
JSON is fine, just remove comments and console.log